cyral/resource_cyral_repository_user_account.go

```diff
@@ -0,0 +1,414 @@
+package cyral
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/cyralinc/terraform-provider-cyral/client"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var allAuthSchemes = []string{
+	"aws_iam",
+	"aws_secrets_manager",
+	"cyral_storage",
+	"hashicorp_vault",
+	"environment_variable",
+	"kubernetes_secret",
+	"gcp_secrets_manager",
+}
+
+type AuthScheme struct {
+	AWSIAM              *AuthSchemeAWSIAM              `json:"awsIAM"`
+	AWSSecretsManager   *AuthSchemeAWSSecretsManager   `json:"awsSecretsManager"`
+	CyralStorage        *AuthSchemeCyralStorage        `json:"cyralStorage"`
+	HashicorpVault      *AuthSchemeHashicorpVault      `json:"hashicorpVault"`
+	EnvironmentVariable *AuthSchemeEnvironmentVariable `json:"environmentVariable"`
+	KubernetesSecret    *AuthSchemeKubernetesSecret    `json:"kubernetesSecret"`
+	GCPSecretManager    *AuthSchemeGCPSecretManager    `json:"gcpSecretManager"`
+}
+
+type AuthSchemeAWSIAM struct {
+	RoleARN string `json:"roleARN"`
+}
+
+type AuthSchemeAWSSecretsManager struct {
+	SecretArn string `json:"secretARN"`
+}
+
+type AuthSchemeCyralStorage struct {
+	Password string `json:"password"`
+}
+
+type AuthSchemeHashicorpVault struct {
+	Path                 string `json:"path"`
+	IsDynamicUserAccount bool   `json:"isDynamicUserAccount"`
+}
+
+type AuthSchemeEnvironmentVariable struct {
+	VariableName string `json:"variableName"`
+}
+
+type AuthSchemeKubernetesSecret struct {
+	SecretName string `json:"secretName"`
+	SecretKey  string `json:"secretKey"`
+}
+
+type AuthSchemeGCPSecretManager struct {
+	SecretName string `json:"secretName"`
+}
+
+type ApprovalConfig struct {
+	AutomaticGrant            bool `json:"automaticGrant"`
+	MaxAutomaticGrantDuration int  `json:"maxAutomaticGrantDuration"`
+}
+
+type UserAccountConfig struct {
+	Approval *ApprovalConfig `json:"approvalConfig"`
+}
+
+type UserAccountResource struct {
+	UserAccountID    string             `json:"userAccountID"`
+	Name             string             `json:"name"`
+	AuthDatabaseName string             `json:"authDatabaseName"`
+	AuthScheme       *AuthScheme        `json:"authScheme"`
+	Config           *UserAccountConfig `json:"config"`
+}
+
+type CreateUserAccountResponse struct {
+	UserAccountID string `json:"userAccountID"`
+}
+
+func (resp *CreateUserAccountResponse) WriteToSchema(d *schema.ResourceData) error {
+	d.SetId(resp.UserAccountID)
+	return nil
+}
+
+func (response *UserAccountResource) WriteToSchema(d *schema.ResourceData) error {
+	d.SetId(response.UserAccountID)
```

Comment:
I believe that this function will need to write all the details of the `UserAccountResource` you just read into the `d` map thingy. If you don't and you change something about this user account in the UI, then Terraform won't know about it and a subsequent `terraform apply` will wrongly think that nothing changed, and thus things will diverge.